opensquirrel/ir.py

```diff
@@ -387,3 +487,7 @@ def __repr__(self) -> str:
     def accept(self, visitor: IRVisitor) -> None:
         for statement in self.statements:
             statement.accept(visitor)
+
+
+# Type Aliases
+AxisLike = Union[ArrayLike, Axis]
```

Comment:
Just a comment, a double check. Not sure how portable across Python versions this `Union` is.

Reply:
Type aliases are a hot mess, currently these are the preferred ways to do it based on python version:
Python312 (not yet compatible with `mypy`):
```python
type AxisLike = ArrayLike | Axis
```
Python 310 and 311 (compatible with `mypy`):
```python
from typing import TypeAlias

AxisLike: TypeAlias = ArrayLike | Axis
```
Python 38 and 39 (compatible with `mypy`):
```python
from typing import Union

AxisLike = Union[ArrayLike, Axis]
```

Usually I would take the newest version and hide it behind an ```if TYPE_CHECKING:``` block. However, we need this alias in other modules as well, were we then would also need such blocks. Instead I chose the the python 38/39 version of type aliasing as it currently works for all python versions we support. When we drop support for python 39 (Okt-2025) we should update this accordingly.

Reply:
Thanks for the detailed explanation.

Maybe worth opening an issue to log a future needed change? Something like `Update type aliases when we drop support for Python 3.9 (October 2025)`?

Reply:
Made issues for the end of life of python3.8 (#227) and python3.9 (#228)

Reply:
Thanks!